recurly/response.go

```diff
@@ -16,8 +16,12 @@ type (
 		// Errors holds an array of validation errors if any occurred.
 		Errors []Error
 
+		// Transaction holds the transaction returned with a transaction error.
+		// This will only be populated when a TransactionError is present.
+		Transaction Transaction
```

Comment:
When calling `CreateSubscription` if the payment fails, Recurly does not return a subscription, like it would if it was successful. It only returns `transaction`, `transaction_error`, and `error`. If it is successful, it returns a `subscription` without any transaction information. It's not great API structure. A `transaction` would not be unmarshalled as part of the struct provided by each API method (subscription in this case), which is why I added it here. What do you think?